src/ValueFormatters/QuantityFormatter.php

```diff
@@ -150,43 +172,50 @@ protected function formatQuantityValue( QuantityValue $quantity ) {
 	/**
 	 * @since 0.6
 	 *
-	 * @param QuantityValue $quantity
+	 * @param UnboundedQuantityValue $quantity
 	 *
 	 * @return string Text
 	 */
-	protected function formatNumber( QuantityValue $quantity ) {
-		$roundingExponent = $this->getRoundingExponent( $quantity );
+	protected function formatNumber( UnboundedQuantityValue $quantity ) {
+		$roundingMode = $this->options->getOption( self::OPT_APPLY_ROUNDING );
+		$roundingExponent = $this->getRoundingExponent( $quantity, $roundingMode );
 
 		$amount = $quantity->getAmount();
 		$roundedAmount = $this->decimalMath->roundToExponent( $amount, $roundingExponent );
 		$formatted = $this->decimalFormatter->format( $roundedAmount );
 
-		$margin = $this->formatMargin( $quantity->getUncertaintyMargin(), $roundingExponent );
-		if ( $margin !== null ) {
-			// TODO: use localizable pattern for constructing the output.
-			$formatted .= '±' . $margin;
+		if ( $quantity instanceof QuantityValue ) {
+			// TODO: strip trailing zeros from margin
+			$margin = $this->formatMargin( $quantity->getUncertaintyMargin(), $roundingExponent );
+			if ( $margin !== null ) {
+				// TODO: use localizable pattern for constructing the output.
+				$formatted .= '±' . $margin;
+			}
 		}
 
 		return $formatted;
 	}
 
 	/**
-	 * Returns the rounding exponent based on the given $quantity
-	 * and the @see QuantityFormatter::OPT_APPLY_ROUNDING option.
+	 * Returns the rounding exponent based on the given $quantity.
 	 *
-	 * @param QuantityValue $quantity
+	 * @param UnboundedQuantityValue $quantity
+	 * @param bool|int $roundingMode The rounding exponent, or true for rounding to the order of
+	 *        uncertainty (significant digits) of a QuantityValue, or false to not apply
+	 *        rounding (that is, round to all digits).
 	 *
 	 * @return int
 	 */
-	private function getRoundingExponent( QuantityValue $quantity ) {
-		if ( $this->options->getOption( self::OPT_APPLY_ROUNDING ) === true ) {
-			// round to the order of uncertainty
+	private function getRoundingExponent( UnboundedQuantityValue $quantity, $roundingMode ) {
+		if ( is_int( $roundingMode ) ) {
+			// round to the given exponent
+			return $roundingMode;
```

Comment:
Support for integers in a boolean option that's called "apply rounding"? Sorry, not convinced. Especially because the numbers are base-10 exponents, and conflict with what lower and upper bound are meant to do.

Again, this looks like a new feature that **must** be in it's own pull request.